### DISCUSSION TITLE
TMDL support for Workspace Mode

### DISCUSSION THREAD

**Discussion by tsy0716** [TabularEditor/TabularEditor3#1262] [Ideas]
### Describe the solution you'd like

Are we getting TMDL support for Workspace Mode soon? The current workspace mode only supports model.bim and database.json. 

### Goals

1.
2.

**Comment by otykier** [comment]
This should already work today. When you create a **new model** in Workspace Mode, you are shown the following prompt upon saving it for the first time:

![image](https://github.com/TabularEditor/TabularEditor3/assets/8976200/c36467e1-3e78-416c-9119-af645950babb)

By choosing the "TMDL (Preview)" option, the model gets saved as TMDL on disk.

For **existing models** in workspace mode, do this:

1. First must make sure that the **Serialization mode** setting is set to "TMDL (Preview)" under **Tools > Preferences > Save-to-folder**. 
2. Then, open an existing model (doesn't matter whether it's already using workspace mode or not) and use the **File > Save to Folder** option. This will now save the model using TMDL. 
3. Close the model (**File > Close Model**) and reopen it from the TMDL you just saved.
4. You should now be prompted for whether you want to add a workspace database. Hit "Yes", then create a new workspace database or reuse the workspace database you were previously using.
